cluster-app-values-schema/README.md

```diff
@@ -0,0 +1,383 @@
+  "type": "string",
+  "title": "Auto refresh",
+  "enum": ["active", "inactive"]
+}
+```
+
+However, if the string values are not considered self-explanatory, the `oneOf` method can be used to provide user-friendly labels in addition to the axctual value. Here, `oneOf` is an array of tuples, in which `title` contains the user-friendly description and `const` provides the value.
+
+```json
+"imagePullPolicy": {
+  "type": "string",
+  "title": "Pull policy",
+  "description": "Whether the container image should be pulled from a registry",
+  "oneOf": [
+    {"const": "IfNotPresent", "title": "Pull image if not present in the node"},
+    {"const": "Always", "title": "Always pull up-to-date image"},
+    {"const": "Never", "title": "Never pull image, use image available locally"}
+  ]
+}
+```
+
+### R13: Avoid recursion {#r13}
+
+The JSON Schema keywords `dynamicRef`, `dynamicAnchor`, and `recursiveRef` MUST NOT be used.
+
+### R14: Avoid logical constructs using `if`, `then`, `else` {#r14}
+
+The JSON Schema keywords `if`, `then`, and `else` MUST NOT be used.
+
+### R15: Avoid `unevaluatedProperties`, `unevaluatedItems` {#r15}
+
+The JSON Schema keywords `unevaluatedProperties` , `unevaluatedItems` MUST NOT be used.
+
+### R16: Array items and tuple validation {#r16}
+
+All array items MUST be of the same type.
+
+The JSON Schema keywords `contains`, `additionalItems` and `prefixItems` MUST NOT be used.
+
+### R17: Common schema structure {#r17}
+
+For cluster apps, we aim to use a common schema structure among the providers we support. This structure accounts for the necessary differences that can occur between providers, as some concepts and implementations are provider-specific.
+
+Each cluster app values schema MUST offer the following root level properties:
+
+| Property name | Property type | Description |
+|-|-|-|
+| `metadata` | object | Descriptive settings like name, description, cluster labels (e. g. service priority), owner organization |
+| `connectivity` | object | Settings related to connectivity and networking, and defining how the cluster reaches the outside world and how it can be reached. |
+| `controlPlane` | object | Configuration of the cluster's control plane, Etcd, API server and more. |
+| `nodePools` | array | Configuration of node pools (groups of worker nodes), regardless of their implentation flavour. |
+
+In addition, the cluster app values schema SHOULD offer the following root level properties:
+
+| Property name | Property type | Description |
+|-|-|-|
+| `internal` | object | Settings which are not supposed to be configured by end users, and which won't be exposed via user interfaces. Also experimental features that undergo schema changes. |
+| `providerSpecific` | object | Configuration specific to the infrastructure provider. |
+
+For compatibility reasons, the schema MAY have the following properties in the root level:
+
+- `managementCluster`
+- `baseDomain`
+- `provider`
+- `cluster-shared`
```

Comment:
We can link to https://github.com/giantswarm/cluster-shared/tree/main/helm/cluster-shared as explanation